Extensions/timestamps.js

```diff
@@ -108,37 +107,58 @@ XKit.extensions.timestamps = new Object({
 
 		if (XKit.interface.where().search) {
 			this.in_search = true;
-			XKit.tools.add_css(`
-				.xtimestamp-in-search {
-					position: absolute;
-					top: 32px;
-					color: rgb(168,177,184);
-					font-size: 10px;
-				}`, "timestamps");
 		}
 
 		if (this.preferences.format.value === "") {
 			this.preferences.format.value = "MMMM Do YYYY, h:mm:ss a";
 		}
 
 		this.check_quota();
+
+		$(document).on("click", ".xkit-timestamp-failed-why", function() {
+			XKit.window.show("Timestamp loading failed.", "This might be caused by several reasons, such as the post being removed, becoming private, or the Tumblr server having a problem that it can't return the page required by XKit to load you the timestamp.", "error", "<div id=\"xkit-close-message\" class=\"xkit-button\">OK</div></div>");
+		});
+
+		if (XKit.page.react) {
+			XKit.css_map.getCssMap()
+			.then(() => {
+				this.posts_class = XKit.css_map.keyToCss("post");
+				this.reblogs_class = XKit.css_map.keyToCss("reblog");
+				this.reblog_headers_class = XKit.css_map.keyToCss("reblogHeader");
+				this.blog_link_class = XKit.css_map.keyToCss("blogLink");
+			});
+
+			if (this.preferences.posts.value || (this.preferences.inbox.value && XKit.interface.where().inbox)) {
```

Comment:
the inbox doesn't have a react page yet, so this second condition will never activate/is redundant for now

```suggestion
			if (this.preferences.posts.value) {
```

Reply:
Shouldn't we keep this here for when the react dash does get an inbox? Less work for the future, right?

Reply:
i guess so. my line of thinking was that we don't know if `react_add_timestamps()` will work on the react inbox, but if it doesn't or produces erroneous behaviour, then that still applies to everyone that has all post timestamps enabled anyway...